### ISSUE TITLE
[Metricbeat] SNMP input or module

### ISSUE THREAD

**Issue by markwalkom** [elastic/beats#12798]
**Describe the enhancement:** It'd be super awesome to support hardware or systems that only have SNMP metrics exposed, or to collect information that other beats may not have access to.

**Describe a specific use case for the enhancement or feature:** I use Ubiquiti networking gear at home and it'd be great to point a beat at an SNMP endpoint, with a list of MIBs or a reference to a file with them(?), to collect data.

I have compiled metricbeat for some of their gera (as it's ARM based, which we don't officially support), so this seems like an alternative method to collecting host metrics.

There is at least [one community beat](https://github.com/isalgueiro/otilio) that does something like this. And I am also aware that there is an [SNMP input](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-snmp.html) in Logstash.

**Comment by fearful-symmetry** [comment]
This has been a reoccurring request, and It would be super neat if we could actually do it. I also use UniFi gear in my home lab, and I've poked around their MIBs.

It seems like the most 'natural' way to do this is to have a SNMP module, and then provide different metricsets for MIB ranges, i.e, a metricset for `iso(1).org(3).dod(6).internet(1).mgmt(2).mib-2(1).ip(4)` one for `.iso(1).org(3).dod(6).internet(1).mgmt(2).mib-2(1).interfaces(2)`, one for the special UniFi MIBs, etc. This would actually be great fodder for the new "lightweight" modules that we're exploring, since most of the code would be module level, with individual metricsets mostly being configs for the various MIBs.

**Comment by hrvojep** [comment]
I've got Australian Gov Department of Human Services asking for this as well. It would be good if we could do this with beats. In this case they use SMARTS (from ECM/VmWare) for monitoring of routers,switches etc via SNMP, but they're looking to us to move away from SMARTS.

**Comment by enotspe** [comment]
I would add to the request other types of remote metric monitoring

[Streaming Telemetry](http://www.openconfig.net/projects/telemetry/)
[WMI](https://github.com/eskibars/wmibeat)
[SSH](http://www.rtop-monitor.org/)

**Comment by loganmc10** [comment]
Yes this would be really nice. Right now I do this using Prometheus+snmp_exporter. I then send it to metricbeat using remote_write (metricbeat is a "sidecar" docker container).

However it's a bit of a pain to graph the metrics in Kibana (I'm still trying to figure this part out). It would be nice if it was a little more integrated by default into Kibana the way that regular host metrics are currently.

**Comment by enotspe** [comment]
Altough I totally love logstash, I have ended up using [Zabbix + Elastic](https://www.zabbix.com/documentation/current/manual/appendix/install/elastic_search_setup), [snmp input plugin](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-snmp.html) is just too basic. 

You kind of get the best of both worlds: NMS functions + super custom dashboards with Kibana and Canvas.

**Comment by zez3** [comment]
Can someone please reopen this
@cmacknz ?

**Comment by zez3** [comment]
Thie might interess the @observability team as well
@andrewvc

**Comment by zez3** [comment]
@felixbarny got any opinions on this?

**Comment by Erikg346** [comment]
+1
Any update on this? This would be so useful! Especially from migrating away from Logstash inputs

**Comment by Erikg346** [comment]
Hey @felixbarny OpenTelemetry’s SNMP receiver is an option, but that's the same as prometheus snmp exporter or the other methods mentioned in this thread.

Ideally, SNMP polling should be native to Elastic. Other platforms already support SNMP for agentless or lightweight monitoring, and it’s a key gap in Elastic's infrastructure monitoring. Adding a separate collector for SNMP creates unnecessary overhead. Appreciate the suggestion!

**Comment by strawgate** [comment]
We’re on a path to integrate OTel-based collection natively into the Elastic ingestion architecture.

The plan is to allow running OTel receivers, Beats, and Elastic Integrations side-by-side within a single collector.

**Comment by daniela-elastic** [comment]
Would the customer be interested in using an [OTel integration for SNMP](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/snmpreceiver)?
Context - Elastic's observability strategy is to go all in on Opentelemetry. Our aim is to build new integrations using the [existing upstream OTel receivers](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver). If there are any missing receivers, we will contribute upstream, based on customer demand. To help customers get value out of the collected signals, we will provide out of the box content like dashboards and in the near future, OOTB alerts and SLOs (aka Content packs).

**Comment by daniela-elastic** [comment]
As another alternative to OTel integration for SNMP, would customers consider using the existing [Logstash plugin for SNMP](https://www.elastic.co/docs/reference/logstash/plugins/plugins-inputs-snmp)?

**Comment by Erikg346** [comment]
Any update on snmp reciever being added?

**Comment by daniela-elastic** [comment]
This is in our backlog. We do not currently have an ETA to share but we will update as and when we do.

**Comment by herrBez** [comment]
The SNMP receiver is already integrated into Elastic-Agent (https://www.elastic.co/docs/reference/edot-collector/components#list-of-components), meaning that in standalone mode it can be used today. What we are missing is an OTEL Input package to run the receiver in Fleet-Mode.
